docs/architecture/blockchain.md

```diff
@@ -0,0 +1,70 @@
+
+`Block`s represent the delta of the global [state](state.md) between two time periods. The current global state can be derived by applying all the blocks to the genesis state.
+
+Miden's blockchain protocol aims for the following:
+
+- **Proven transactions**: All included transactions have already been proven and verified when they reach the block.
+- **Fast genesis syncing**: New nodes efficiently sync to the network through a multi-step process:
+
+  1. Download historical `Block`s from genesis to the present.
+  2. Verify zero-knowledge proofs for all `Block`s.
+  3. Retrieve current state data (accounts, notes, and nullifiers).
+  4. Validate that the downloaded state matches the latest `Block`'s state commitment.
+
+This approach enables near-instant blockchain syncing by verifying `Block` proofs rather than re-executing individual transactions, resulting in exponentially faster performance. Consequently, state sync is dominated by the time needed to download the data.
+
+<p style="text-align: center;">
+    <img src="../img/architecture/blockchain/execution.png" style="width:70%;" alt="Execution diagram"/>
+</p>
+
+## Batch production
+
+A Miden block consist of multiple transaction batches. This enables recursive proving and also allows transactions to be processed concurrently as batches with no overlap in accounts and notes can be built in parallel. Miden will have multiple batch producers operating simultaneously and, together with offchain/client-side transaction proving, this massively reduces the work the network is required to do.
+
+The purpose of this scheme is to produce a single proof that attests to the validity of a number of transactions. This is achieved by recursively verifying each transaction proof within the Miden VM.
+
+<p style="text-align: center;">
+    <img src="../img/architecture/blockchain/batching.png" style="width:50%;" alt="Batch diagram"/>
+</p>
+
+The batch producer aggregates transactions sequentially by verifying their proofs and state transitions are correct. More specifically, the batch producers ensures:
+
+1. **Ordering of transactions**: If several transactions within the same batch affect a single account, the correct ordering must be enforced. For example, if `Tx1` and `Tx2` both describe state changes of account `A`, then the batch kernel must verify them in the order: `A -> Tx1 -> A' -> Tx2 -> A''`.
+2. **Prevention of double spending and duplicate notes**: The batch producer must ensure the uniqueness of all notes across transactions in the batch. This prevents double spending and avoids the situation where duplicate notes, which would share identical nullifiers, are created. Only one of such duplicate notes can later be consumed, as the nullifier will be marked as spent after the first consumption.
```

Comment:
```suggestion
2. **Prevention of double spending and duplicate notes**: The batch producer must ensure the uniqueness of all notes across transactions in the batch. This prevents double spending and avoids the situation where duplicate notes, which would share identical nullifiers, are created. Only one of such duplicate notes could later be consumed, as the nullifier will be marked as spent after the first consumption.
```

Nit: "could" rather than "can" to show this is a theoretical scenario.